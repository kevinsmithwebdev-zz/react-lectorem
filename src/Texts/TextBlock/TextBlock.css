
.TextBlock {
  text-align: left;
  flex: 1;
}
