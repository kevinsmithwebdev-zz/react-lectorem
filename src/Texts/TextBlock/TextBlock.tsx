import React from 'react';
import Sentence from '../../common/Sentence/Sentence';
import './TextBlock.css';
import { SentenceInterface, ConfigurationInterface } from '../../interfaces/index';

interface TextBlockInterface {
  block: {
    sentences: SentenceInterface[],
  }
  configuration: ConfigurationInterface,
  idx: number,
  readTime: number,
  showModal: Function,
}

const TextBlock: React.FC<TextBlockInterface> = ({
  block: { sentences },
  configuration,
  idx,
  readTime,
  showModal,
}) => (
  <div className='TextBlock' key={ idx }>
    {
      sentences?.map(
        /* istanbul ignore next */ (sentence, idx2) => (
          <Sentence
            configuration={ configuration }
            sentence={ sentence }
            key={ idx2.toString() }
            readTime={ readTime }
            showModal={ showModal }
          />
        )
      )
    }
  </div>
);

export default TextBlock;
