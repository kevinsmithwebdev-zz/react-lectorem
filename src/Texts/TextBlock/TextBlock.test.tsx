import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TextBlock from './TextBlock';
import storyData from '../../fixtureData/story.data';

describe('TextBlock', () => {
  describe('the main component render', () => {
    const { sentences } = storyData.paragraphs[0];
    const configuration = {
      isTranslationShown: true,
      isTranslationBelow: true,
      isBySentence: true,
    };
    const renderer = ShallowRenderer.createRenderer();
    describe('with data', () => {
      const props = {
        block: { sentences },
        configuration,
        idx: 127,
        readTime: 3.1459,
        showModal: () => {},
      };

      it('should render and match snapshot', () => {
        renderer.render(<TextBlock {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });
});
