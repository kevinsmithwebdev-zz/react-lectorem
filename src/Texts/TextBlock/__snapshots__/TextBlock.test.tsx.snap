// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock the main component render with data should render and match snapshot 1`] = `
<div
  className="TextBlock"
>
  <Sentence
    configuration={
      Object {
        "isBySentence": true,
        "isTranslationBelow": true,
        "isTranslationShown": true,
      }
    }
    readTime={3.1459}
    sentence={
      Object {
        "lects": Array [
          Object {
            "end": 5.511,
            "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
            "start": 4.424,
            "text": "Frère",
            "translation": "Brother",
          },
          Object {
            "end": 6.533,
            "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
            "start": 5.51,
            "text": "Jacques",
            "translation": "James",
          },
          Object {
            "text": ".",
          },
        ],
        "sentenceTranslation": "Brother James.",
      }
    }
    showModal={[Function]}
  />
  <Sentence
    configuration={
      Object {
        "isBySentence": true,
        "isTranslationBelow": true,
        "isTranslationShown": true,
      }
    }
    readTime={3.1459}
    sentence={
      Object {
        "lects": Array [
          Object {
            "end": 7.683,
            "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
            "start": 6.618,
            "text": "Frère",
            "translation": "Brother",
          },
          Object {
            "end": 8.79,
            "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
            "start": 7.68,
            "text": "Jacques",
            "translation": "James",
          },
          Object {
            "text": ".",
          },
        ],
        "sentenceTranslation": "Brother James.",
      }
    }
    showModal={[Function]}
  />
</div>
`;
